nujo/autodiff/_functions/_trigonometric.py

```diff
@@ -0,0 +1,71 @@
+from numbers import Number
+from typing import List, Union
+
+from numpy import cos, ndarray, sin, tan
+
+from nujo.autodiff.function import Function
+from nujo.autodiff.tensor import Tensor
+
+__all__ = ['_Sin', '_Cos', '_Tan']
+
+# ====================================================================================================
+
+
+class _Sin(Function):
+    ''' Differable sine function
+
+    sin(X) = (e^iX - e^-iX) / 2i
+
+    '''
+    def __init__(self, input: Union[Tensor, ndarray, List[Number], Number]):
+        super(_Sin, self).__init__(input)
+
+    def forward(self) -> ndarray:
+        return sin(self.children[0].value)
+
+    def backward(self) -> Function.T:
+        # cos(X)
+        return Tensor(_Cos(self.children[0].value)())
+
+
+# ====================================================================================================
+
+
+class _Cos(Function):
+    ''' Differable cosine function
+
+    cos(X) = (e^iX + e^-iX) / i
+
+    '''
+    def __init__(self, input: Union[Tensor, ndarray, List[Number], Number]):
+        super(_Cos, self).__init__(input)
+
+    def forward(self) -> ndarray:
+        return cos(self.children[0].value)
+
+    def backward(self) -> Function.T:
+        # -sin(X)
+        return Tensor(-(_Sin(self.children[0].value)()))
```

Comment:
Same, `backward` signature - incorrect and don't use `_Sin` (nujo Function) here, use numpy's `sin`